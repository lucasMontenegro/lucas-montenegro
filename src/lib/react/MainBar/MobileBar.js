import { makeStyles } from "@material-ui/core/styles"
import useTranslation from "lib/react/useTranslation"
import useMobileNav from "./useMobileNav"
import React from "react"
import AppBar from "@material-ui/core/AppBar"
import Toolbar from "@material-ui/core/Toolbar"
import Tooltip from "@material-ui/core/Tooltip"
import IconButton from "@material-ui/core/IconButton"
import { useAppDrawerOpener } from "lib/react/AppDrawer"
//import { useDashboardOpener } from "lib/react/Dashboard"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
const useStyles = makeStyles(theme => ({
  appBar: {
    color: theme.palette.text.primary,
    backgroundColor: theme.palette.background.paper,
    top: `auto`,
    bottom: 0,
  },
  toolbar: {
    "& > button.icon-button": {
      minWidth: theme.spacing(6),
    },
    "& > div.flexible-space": {
      flexBasis: `100%`,
      flexShrink: 1,
    },
  },
}), { name: `lib-react-main_bar-mobile_bar` })
export default function MobileBar () {
  const classes = useStyles()
  const t = useTranslation()
  const mobileNav = useMobileNav(t)
  return (
    <AppBar className={classes.appBar}>
      <Toolbar className={classes.toolbar}>
        <Tooltip
          title={t({
            en: () => `Open dashboard`,
            es: () => `Abrir panel de control`,
          })}
        >
          <IconButton
            className="icon-button"
            edge="start"
            aria-controls="lib-react-app_drawer"
            onClick={useAppDrawerOpener()}
          >
          {/*
            aria-controls="lib-react-dashboard"
            onClick={useDashboardOpener()}
          >
          */}
            <FontAwesomeIcon icon={[`fas`, `bars`]} />
          </IconButton>
        </Tooltip>
        <div className="flexible-space" />
        <Tooltip
          title={t({
            en: () => `Open quick navigation`,
            es: () => `Abrir navegación rápida`,
          })}
        >
          <IconButton
            edge="end"
            className="icon-button"
            aria-controls="lib-react-main_bar-mobile_nav"
            onClick={mobileNav.open}
          >
            <FontAwesomeIcon icon={[`fas`, `caret-up`]} />
          </IconButton>
        </Tooltip>
        {mobileNav.node}
      </Toolbar>
    </AppBar>
  )
}