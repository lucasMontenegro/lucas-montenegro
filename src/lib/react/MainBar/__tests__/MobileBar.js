import renderer from "react-test-renderer"
import React from "react"
import MobileBar from "../MobileBar"
jest.mock(`@material-ui/core/styles`, () => ({
  __esModule: true,
  makeStyles: (styles, options) => () => {
    const classes = styles({
      palette: {
        text: { primary: `theme.palette.text.primary` },
        background: { paper: `theme.palette.background.paper` },
      },
      spacing: n => 8*n,
    })
    return Object.keys(classes).reduce((obj, key) => {
      obj[key] = `${options.name}-${key}(${JSON.stringify(classes[key]).replace(/"/gm, `'`)})`
      return obj
    }, {})
  },
}))
jest.mock(`lib/react/useTranslation`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: () => source => (
      <div className="translation">
        {Object.keys(source).map(languageCode => (
          <div key={languageCode} className={languageCode}>{source[languageCode]()}</div>
        ))}
      </div>
    ),
  }
})
jest.mock(`../useMobileNav`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: t => ({
      open: () => `useMobileNav().open()`,
      node: (
        <div className="MobileNav">
          {t({
            en: () => `EN MobileNav`,
            es: () => `ES MobileNav`,
          })}
        </div>
      ),
    }),
  }
})
jest.mock(`@material-ui/core/AppBar`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: props => <div {...props} className={`AppBar ${props.className}`} />,
  }
})
jest.mock(`@material-ui/core/Toolbar`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: props => <div {...props} className={`Toolbar ${props.className}`} />,
  }
})
jest.mock(`@material-ui/core/Tooltip`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="Tooltip" /> }
})
jest.mock(`@material-ui/core/IconButton`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: props => (
      <div {...props} className={`IconButton ${props.className}`} onClick={props.onClick()} />
    ),
  }
})
jest.mock(`lib/react/AppDrawer`, () => ({
  __esModule: true,
  useAppDrawerOpener: () => () => `useAppDrawerOpener()()`,
}))
jest.mock(`lib/react/fontAwesome`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    FontAwesomeIcon: props => <i {...props} className="FontAwesomeIcon" />,
  }
})
describe(`../MobileBar`, () => {
  it(`should use the right versions of its dependencies`, () => {
    expect(jestUtils.getDependencies([`@material-ui/core`, `react`])).toMatchSnapshot()
  })
  describe(`<MobileBar />`, () => {
    it(`should render`, () => {
      expect(renderer.create(<MobileBar />).toJSON()).toMatchSnapshot()
    })
  })
})