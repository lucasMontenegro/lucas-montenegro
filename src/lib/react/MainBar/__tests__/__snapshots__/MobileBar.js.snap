// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../MobileBar <MobileBar /> should render 1`] = `
<div
  className="AppBar lib-react-main_bar-mobile_bar-appBar({'color':'theme.palette.text.primary','backgroundColor':'theme.palette.background.paper','top':'auto','bottom':0})"
>
  <div
    className="Toolbar lib-react-main_bar-mobile_bar-toolbar({'& > button.icon-button':{'minWidth':48},'& > div.flexible-space':{'flexBasis':'100%','flexShrink':1}})"
  >
    <div
      className="Tooltip"
      title={
        <div
          className="translation"
        >
          <div
            className="en"
          >
            Open dashboard
          </div>
          <div
            className="es"
          >
            Abrir panel de control
          </div>
        </div>
      }
    >
      <div
        aria-controls="lib-react-app_drawer"
        className="IconButton icon-button"
        edge="start"
        onClick="useAppDrawerOpener()()"
      >
        <i
          className="FontAwesomeIcon"
          icon={
            Array [
              "fas",
              "bars",
            ]
          }
        />
      </div>
    </div>
    <div
      className="flexible-space"
    />
    <div
      className="Tooltip"
      title={
        <div
          className="translation"
        >
          <div
            className="en"
          >
            Open quick navigation
          </div>
          <div
            className="es"
          >
            Abrir navegación rápida
          </div>
        </div>
      }
    >
      <div
        aria-controls="lib-react-main_bar-mobile_nav"
        className="IconButton icon-button"
        edge="end"
        onClick="useMobileNav().open()"
      >
        <i
          className="FontAwesomeIcon"
          icon={
            Array [
              "fas",
              "caret-up",
            ]
          }
        />
      </div>
    </div>
    <div
      className="MobileNav"
    >
      <div
        className="translation"
      >
        <div
          className="en"
        >
          EN MobileNav
        </div>
        <div
          className="es"
        >
          ES MobileNav
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`../MobileBar should use the right versions of its dependencies 1`] = `
Object {
  "@material-ui/core": "^4.5.0",
  "react": "^16.10.1",
}
`;
